src/IECoreMaya/SceneShapeSubSceneOverride.cpp

```diff
@@ -237,378 +185,415 @@ class Expandulator
+
+VertexBufferPtr createVertexBuffer( const MVertexBufferDescriptor &descriptor, const V3fVectorData *data )
 {
-	GeometryDataPtr geometryData( new GeometryData() );
+	VertexBufferPtr buffer( new MVertexBuffer( descriptor ) );
 
-	if( !key.object )
+	size_t numEntries = data->readable().size();
+	void *bufferData = buffer->acquire( numEntries, true );
+	if( bufferData && buffer )
 	{
-		fillBoundData( key.bounds, geometryData.get() );
-		cost = computeGeometryDataCost( geometryData.get() );
-		return geometryData;
+		memcpy( bufferData, data->baseReadable(), sizeof( float ) * 3 * numEntries );
+		buffer->commit( bufferData );
 	}
-
-	switch( static_cast<IECoreScene::TypeId>( key.object->typeId() ) )
+	else
 	{
-	case IECoreScene::TypeId::MeshPrimitiveTypeId :
-		fillMeshData( key.object.get(), key.descriptorList, geometryData.get() ); break;
-	case IECoreScene::TypeId::PointsPrimitiveTypeId :
-		fillPointsData( key.object.get(), geometryData.get() ); break;
-	case IECoreScene::TypeId::CurvesPrimitiveTypeId :
-		fillCurvesData( key.object.get(), geometryData.get() ); break;
-	default :
-		// we should never get here as only the above types are supported, see `objectCanBeRendered`.
-		break;
+		msg( Msg::Error, "SceneShapeSubSceneOverride::bufferGetter", ( boost::format( "Memory acquisition for %s buffer failed." ) % descriptor.semanticName() ).str() );
 	}
 
-	cost = computeGeometryDataCost( geometryData.get() );
-
-	return geometryData;
+	return buffer;
 }
 
-using GeometryDataCache = IECore::LRUCache<IECore::MurmurHash, GeometryDataPtr, IECore::LRUCachePolicy::Parallel, GeometryDataCacheGetterKey>;
-
-GeometryDataCache &geometryDataCache()
+VertexBufferPtr createVertexBuffer( const MVertexBufferDescriptor &descriptor, const V2fVectorData *data )
 {
-	static GeometryDataCache g_geometryDataCache( geometryGetter, memoryLimit() * 0.25 );
-	return g_geometryDataCache;
+	VertexBufferPtr buffer( new MVertexBuffer( descriptor ) );
+
+	size_t numEntries = data->readable().size();
+	void *bufferData = buffer->acquire( numEntries, true );
+	if( bufferData && buffer )
+	{
+		memcpy( bufferData, data->baseReadable(), sizeof( float ) * 2 * numEntries );
+		buffer->commit( bufferData );
+	}
+	else
+	{
+		msg( Msg::Error, "SceneShapeSubSceneOverride::bufferGetter", "Memory acquisition for uv buffer failed." );
+	}
+
+	return buffer;
 }
 
-////////////////////////////////////////////////////////////////////////////////////////////
-// Maya Buffer Cache
-////////////////////////////////////////////////////////////////////////////////////////////
+using CacheEntry = boost::variant<VertexBufferPtr, ExpandulatorPtr>;
+using CacheEntryPtr = std::shared_ptr<CacheEntry>;
 
 boost::signals2::signal<void (const BufferPtr & )> &bufferEvictedSignal()
 {
 	static boost::signals2::signal<void (const BufferPtr & )> g_bufferEvictedSignal;
 	return g_bufferEvictedSignal;
 }
 
-struct BufferCleanup
+struct CacheCleanup
```

Comment:
I also found this confusing, but I came to the same conclusion as you -- its working, its not a bottleneck or a memory issue as far as we know, so why change it now?

> could we just store a list of the affected render items in the CacheEntry?

Doesn't that involve mutating the CacheEntry each time we use it? I thought that wasn't generally approved of.

Note `expandulator::wireframeIndices()` is doing this currently, and I don't particularly like it, but it does have performance implications otherwise...